modmod: Support images in exercise descriptions

g-braeunlich opened the pull request:
Closes #118

One problem to be discussed:
The current approach does not handle image conflicts (two exercises having images with the same name).
With the current implementation, the images of an exercise with a higher index just would overwrite images from exercises with a lower index.

hdoordt reviewed (changes requested):
Thanks again @g-braeunlich, this looks great in general!

I think we can mitigate the issue with multiple exercises referring to images with the same name, by requiring the markdown to prefix the image name with a `#[modmod:exercise_images]`, like the [others defined here](https://github.com/tweedegolf/teach-rs/blob/63f08316ceb02913ad6b424f91f2b3064624a99c/modmod/src/book.rs#L96-L106) which resolves to a directory that is unique to the exercise within the `images` directory, and place the images in said directory.

This does require a little bit more plumbing

g-braeunlich commented:
Sorry for the late response. But now I think I have implemented your suggestion. It also will create a `images/{chapter_i}/{section_i}/{subsection_i}` folder hierarchy in the output `src` folder.
And I also implemented a check which will print warnings for non existing images / unreferenced images.

g-braeunlich commented:
Would you rather have exceptions instead of warnings?

hdoordt commented:
@g-braeunlich sorry for the delay. I think exceptions would be nicer, as this is something you'd want to be rather 'in your face' when setting up a module. Very nice otherwise!

hdoordt commented on modmod/src/book.rs:
```diff
@@ -30,6 +30,8 @@ pub struct Book<'track> {
     pub chapters: Vec<Chapter<'track>>,
 }
 
+const IMAGE_PLACEHOLDER: &str = "#[modmod:images]";
```
This is nice. Could you make the other placeholders to consts too?

g-braeunlich replied:
Should be done now

hdoordt reviewed (commented):
Very nice so far, thanks already!

hdoordt commented:
With #116, I refactored modmod quite heavily, which will make it non-trivial to merge this PR. I will however rebase this branch and resolve the conflicts once you're happy with your contribution

hdoordt commented:
@g-braeunlich Thanks!
